src/lib/numa.lua

```diff
@@ -28,12 +29,24 @@ function cpu_get_numa_node (cpu)
 end
 
 function has_numa ()
-   local node1 = S.open('/sys/devices/system/node/node1', 'rdonly, directory')
+   local node1 = S.open(node_path..tostring(1), 'rdonly, directory')
    if not node1 then return false end
    node1:close()
    return true
 end
 
+-- Return the first and last CPU ids (as numbers) on the given NUMA node.
+function get_cpus_per_numa_node (node)
```

Comment:
Are you intending this to be a public function?  It would need to be documented.  Also are you sure the list is contiguous?  I think it can be disjoint when hyperthreads are enabled.